mantidproject/mantid
# Fix ISIS_WISHSingleCrystalReduction system test

**Description of work.**

A temporary fix was put into the ISIS_WISHSingleCrystalReduction.WISHProcessVanadiumForNormalisationTest in https://github.com/mantidproject/mantid/pull/31857 to ensure the result was the same on all platforms (the Mac result was different).

This PR implements a permanent fix by change the value of the NumberOfNeighbours parameter to 6 instead of the default value of 8. The centres of the spectra on WISH are arranged in more of a hexagonal way than a standard rectangular grid. This means that the 8th and 9th nearest neighbour can be v close together so that the top 8 neighbours are sometimes different on different platforms due to tiny differences in the floating point precision of the calculated distances. This change aligns the behaviour on all platforms

**To test:**

The Jenkins tests for this PR will verify the system test runs OK on RHEL7. 

I suggest that due to the limited availability of Mac developers we should just check the WISHProcessVanadiumForNormalisationTest system test also passes on one extra platform as well (giving a total of 2) and then review the master nightly jobs after this PR is merged to check the test works on all platforms

One of the updated reference files is quite large so apologies if it takes a while to download while working at home

Fixes #31858

*This does not require release notes* because **it only impacts a system test and results in no functional change to the users**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
